### DISCUSSION TITLE
Customizing Button Configs? [Feature request]

### DISCUSSION THREAD

**Discussion by RendellDraw** [leoetlino/project-restoration#172] [Q&A] [answered]
First of all, I'd like to thank you for coming up with this mod! Now the game is as fun as the OG.

I like how you made more use of the D-pad!
But the thing is, I actually don't mind having to equip masks on the face buttons and the touchscreen slots because I kinda prefer having to deal with inventory management and I feel that it's somewhat part of the gameplay.

I'm wondering if there's a way for me to edit something (I'm not a programmer but Google helped me dabble/mess around in it a bit in the past) so that I could map menu features to the following buttons;

*Masks > D-pad Left

*Map > D-pad Down

*Items > D-pad Right

*View/Pictograph > D-pad Up

*Notebook > Select

*Tatl > Start

I'm hoping you could lead me in the right direction so I could try and find out how I could go about this on my own.

Thanks again!

**Please ignore/delete the Discussion I made about this.

**Comment by leoetlino** [comment, marked as answer]
This isn't really an issue or a feature request, so I have turned this into a discussion.

This is the function you would have to modify:

https://github.com/leoetlino/project-restoration/blob/fbd85ca96037c13b4515fa1477393f3ddd784948/source/rst/main.cpp#L160-L181

You'll also need to disable the fast transform code here: https://github.com/leoetlino/project-restoration/blob/fbd85ca96037c13b4515fa1477393f3ddd784948/source/rst/link.cpp#L52

**Reply by RendellDraw** [reply]
Hi,

I'm trying to run make_releass.sh but git says that it can't find Magikoopa (line 32 something) -- I've tried your fork and the original by putting it in the same path but no joy.

Google's not much of a help with it. 
Do you have any advice as to how I can proceed and what I'm doing wrong?

Thanks a lot by the way!

**Reply by leoetlino** [reply]
Have you compiled Magikoopa?
